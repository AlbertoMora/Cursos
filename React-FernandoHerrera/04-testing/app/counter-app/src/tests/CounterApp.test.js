import CounterApp from "../CounterApp";
import React from "react";
import { shallow } from "enzyme";

describe("Test of CounterApp basics", () => {
    const wrapper = shallow(<CounterApp />);

    test("should match snapshot using default values", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("should show a default value of 100", () => {
        const defValue = 100;
        const wrapper = shallow(<CounterApp value={defValue} />);

        const value = wrapper.find("h2").text();

        expect(value).toBe(defValue.toString());
    });
});

describe("Test of clicking events", () => {
    let wrapper = shallow(<CounterApp />);

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    });
    test("should increase counter value + 1", () => {
        wrapper.find("button").at(0).simulate("click");
        const counterText = wrapper.find("h2").text();

        expect(counterText).toBe("1");
    });

    test("should decrease counter value - 1", () => {
        wrapper.find("button").at(2).simulate("click");
        const counterText = wrapper.find("h2").text();

        expect(counterText).toBe("-1");
    });

    test("should set value to default value", () => {
        const defValue = 100;
        const wrapper = shallow(<CounterApp value={defValue} />);
        wrapper.find("button").at(0).simulate("click");
        wrapper.find("button").at(1).simulate("click");
        const counterText = wrapper.find("h2").text();

        expect(counterText).toBe(defValue.toString());
    });
});
