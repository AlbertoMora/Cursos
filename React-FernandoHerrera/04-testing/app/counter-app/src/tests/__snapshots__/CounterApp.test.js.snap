// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of CounterApp basics should match snapshot using default values 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
